test/e2e/vsphere_volume_diskformat.go

```diff
@@ -0,0 +1,202 @@
+		}
+		if !isNodeLabeled {
+			nodeLabelValue := "vsphere_e2e_" + string(uuid.NewUUID())
+			nodeKeyValueLabel = make(map[string]string)
+			nodeKeyValueLabel["vsphere_e2e_label"] = nodeLabelValue
+			framework.AddOrUpdateLabelOnNode(client, nodeName, "vsphere_e2e_label", nodeLabelValue)
+			isNodeLabeled = true
+		}
+	})
+	AddCleanupAction(func() {
+		if len(nodeLabelValue) > 0 {
+			framework.RemoveLabelOffNode(client, nodeName, "vsphere_e2e_label")
+		}
+	})
+
+	It("verify disk format type - eagerzeroedthick is honored for dynamically provisioned pv using storageclass", func() {
+		By("Invoking Test for diskformat: eagerzeroedthick")
+		invokeTest(client, namespace, nodeName, nodeKeyValueLabel, "eagerzeroedthick")
+	})
+	It("verify disk format type - zeroedthick is honored for dynamically provisioned pv using storageclass", func() {
+		By("Invoking Test for diskformat: zeroedthick")
+		invokeTest(client, namespace, nodeName, nodeKeyValueLabel, "zeroedthick")
+	})
+	It("verify disk format type - thin is honored for dynamically provisioned pv using storageclass", func() {
+		By("Invoking Test for diskformat: thin")
+		invokeTest(client, namespace, nodeName, nodeKeyValueLabel, "thin")
+	})
+})
+
+func invokeTest(client clientset.Interface, namespace string, nodeName string, nodeKeyValueLabel map[string]string, diskFormat string) {
+
+	framework.Logf("Invoking Test for DiskFomat: %s", diskFormat)
+	scParameters := make(map[string]string)
+	scParameters["diskformat"] = diskFormat
+
+	By("Creating Storage Class With DiskFormat")
+	storageClassSpec := getVSphereStorageClassSpec("thinsc", scParameters)
+	storageclass, err := client.StorageV1beta1().StorageClasses().Create(storageClassSpec)
+	defer client.StorageV1beta1().StorageClasses().Delete(storageclass.Name, nil)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("Creating PVC using the Storage Class")
+	pvclaimSpec := getVSphereClaimSpecWithStorageClassAnnotation(namespace, storageclass)
+	pvclaim, err := client.CoreV1().PersistentVolumeClaims(namespace).Create(pvclaimSpec)
+	defer func() {
+		client.CoreV1().PersistentVolumeClaims(namespace).Delete(pvclaimSpec.Name, nil)
+	}()
+
+	By("Waiting for calim to be in bould phase")
+	err = framework.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client, pvclaim.Namespace, pvclaim.Name, framework.Poll, framework.ClaimProvisionTimeout)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Get new copy of the claim
+	pvclaim, err = client.CoreV1().PersistentVolumeClaims(pvclaim.Namespace).Get(pvclaim.Name, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+
+	// Get the bound PV
+	pv, err := client.CoreV1().PersistentVolumes().Get(pvclaim.Spec.VolumeName, metav1.GetOptions{})
+	Expect(err).NotTo(HaveOccurred())
+
+	/*
+		PV is required to be attached to the Node. so that using govmomi API we can grab Disk's Backing Info
+		to check EagerlyScrub and ThinProvisioned property
+	*/
+	By("Creating POD to attach PV to the node")
```

Comment:
super nit: "POD" is not an acronym like "PV" is. Usually lowercase "pod" is seen in messages.